### PR TITLE
Tests: new `Cookie\FormatTest` class

### PR DESCRIPTION
**Pull request by jrfnl**
... with dedicated tests for the `Cookie::__toString()`, `Cookie::format_for_header()` and the `Cookie::format_for_set_cookie()` methods.

Includes removing some test code from the `CookieTest` class which is now covered by the tests in this class.

⚠️ Also take note that, IMO, some of these tests highlight potential bugs. I'd like a second opinion on this though before addressing this.

### CODE REVIEW

**Comment by schlessera** [comment]
> Also take note that, IMO, some of these tests highlight potential bugs. I'd like a second opinion on this though before addressing this.

Should we discuss this here or in a separate issue to unblock and merge this PR ?

**Comment by jrfnl** [comment]
I've opened #743 to discuss the potential bugs.

**Comment by jrfnl** [comment]
I've rebased the PR after the recent merges and added one additional test case to the `dataFormat()` data provider - "Has key, empty value".

**Review by schlessera — APPROVED** [review]

**Pull request merged**
